### ISSUE TITLE
limit with floor problems

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#1880]
Imported from SourceForge on 2024-07-05 10:21:10
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2020-03-16 15:03:52
Original: https://sourceforge.net/p/maxima/bugs/3616

---

limit(floor(atan(x)),x,1) => nounform // NO, should be 0
limit(floor(atan(x)),x,1.0) => 0 // CORRECT

limit(ceiling(abs(x)),x,0) => 1 // CORRECT, even though ceiling(abs(0)) = 0
limit(ceiling(sin(x)^2),x,0) => 1 // CORRECT, even though ceiling(sin(0)^2) = 0
limit(ceiling(1-sin(x)^2),x,0) => 1 // CORRECT
... but ...
limit(ceiling(cos(x)),x,0) => nounform // NO, should be 1
limit(floor(cos(x)),x,0) => UND // NO, should be 0, even though cos(0)=1
limit(ceiling(cos(x)^2),x,0) => nounform // NO, should be 1

Maxima 5.43.0 http://maxima.sourceforge.net
using Lisp SBCL 1.4.14

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:21:12
Created by **[macrakis](https://sourceforge.net/u/macrakis/)** on 2020-03-16 15:04:17
Original: https://sourceforge.net/p/maxima/bugs/3616/#bca7

---

- Description has changed:

Diff:

~~~~

--- old
+++ new
@@ -8,3 +8,6 @@
 limit(ceiling(cos(x)),x,0) =&gt; nounform // NO, should be 1
 limit(floor(cos(x)),x,0) =&gt; UND // NO, should be 0, even though cos(0)=1
 limit(ceiling(cos(x)^2),x,0) =&gt; nounform // NO, should be 1
+
+Maxima 5.43.0 http://maxima.sourceforge.net
+using Lisp SBCL 1.4.14

~~~~

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:21:15
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2020-04-19 20:59:29
Original: https://sourceforge.net/p/maxima/bugs/3616/#3096

---

- **status**: open --> closed
- **assigned_to**: Dan Gildea

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-05 10:21:19
Created by **[dgildea](https://sourceforge.net/u/dgildea/)** on 2020-04-19 20:59:29
Original: https://sourceforge.net/p/maxima/bugs/3616/#83c6

---

Fixed by [b866bd1ba62a1baca4e98be3a7e5a21b831b6cf2]
